deployment/hasura/migrations/Aerie/10_procedural_scheduling/up.sql

```diff
@@ -0,0 +1,155 @@
+alter table scheduler.scheduling_specification_goals
+  add column goal_invocation_id integer generated by default as identity,
+
+  drop constraint scheduling_specification_goals_primary_key,
+  add constraint scheduling_specification_goals_primary_key
+    primary key (goal_invocation_id);
+
+comment on column scheduler.scheduling_specification_goals.goal_invocation_id is e''
+  'The id of a specific goal invocation in the specification. Primary key.';
+
+-- update goal_analysis PK
+alter table scheduler.scheduling_goal_analysis
+  add column goal_invocation_id integer null, -- temp set as nullable so we can insert, made not null by PK constraint below
+
+  drop constraint scheduling_goal_analysis_primary_key;
+
+comment on table scheduler.scheduling_goal_analysis is e''
+  'The analysis of single goal invocation from a scheduling run.';
+
+comment on column scheduler.scheduling_goal_analysis.goal_invocation_id is e''
+  'The associated goal invocation ID.';
+
+update scheduler.scheduling_goal_analysis as sga
+set goal_invocation_id = ssg.goal_invocation_id
+from scheduler.scheduling_request as sr
+       join scheduler.scheduling_specification_goals as ssg
+            on sr.specification_id = ssg.specification_id
+where sr.analysis_id = sga.analysis_id
+  and sga.goal_id = ssg.goal_id;
+
+-- explictly set not null before PKing
+alter table scheduler.scheduling_goal_analysis
+  alter column goal_invocation_id set not null;
+
+alter table scheduler.scheduling_goal_analysis
+  add constraint scheduling_goal_analysis_primary_key
+    primary key (analysis_id, goal_invocation_id);
```

Comment:
These can be done in the same `alter table` statement. This applies to several `alter table set not null` statements.

Reply:
Change has been made